src/components/UserRegisterPage.vue

```diff
@@ -37,37 +58,44 @@
 
               <!-- password -->
               <ValidationObserver>
-                <ValidationProvider name="password" rules="required|min:12" v-slot="{ errors }" vid="repeatPassword">
-                <b-form-group
-                  id="input-group-changepwd-2"
-                  :label="$t('form_password')"
-                  label-for="password"
-                  :description="errors[0]">
-                  <b-form-input
-                    id="password" name="password"
-                    v-model.trim="user.password"
-                    type="password"
-                    maxlength="80"
-                    :description="errors[0]"
-                  ></b-form-input>
-                </b-form-group>
-                </ValidationProvider>
-
-                <ValidationProvider rules="required|confirmed:repeatPassword" v-slot="{ errors }">
-                  <b-form-group
-                    id="input-group-changepwd-3"
-                    :label="$t('form_password_repeat')"
-                    label-for="repeat-password"
-                    :description="errors[0]">
-                    <b-form-input
-                      id="repeat-password" name="repeat-password"
-                      v-model.trim="repeatPassword"
-                      maxlength="80"
-                      type="password" />
-                  </b-form-group>
-                </ValidationProvider>
-              </ValidationObserver><!--  password -->
+                <b-row>
+                  <b-col>
+                    <ValidationProvider name="password"
+                    :rules="{ min: 8, regex: /^(?=.*?[A-Z])(?=.*[a-z])(?=.*[\d])(?=.*[\W_])(?!.*\s).{8,}$/ }"
```

Comment:
Generally it would be good to store long regular expressions in a variable and reference them in template. It would make it easier to read wouldn't it?

Reply:
thanks @theorm I moved the regex to a new module /logic/user